cdap-security/src/main/java/co/cask/cdap/security/server/ExternalAuthenticationServer.java

```diff
@@ -110,11 +110,21 @@ public ExternalAuthenticationServer(CConfiguration configuration, SConfiguration
    * @return InetSocketAddress of server.
    */
   public InetSocketAddress getSocketAddress() {
-    return new InetSocketAddress(address, port);
+    if (!server.isRunning()) {
+      return new InetSocketAddress(address, port);
+    } else {
+      // assumes we only have one connector
+      final Connector connector = server.getConnectors()[0];
+      return new InetSocketAddress(connector.getHost(), connector.getLocalPort());
+    }
   }
 
   @Override
   protected void run() throws Exception {
+    server.start();
+
+    // assumes we only have one connector
+    final Connector connector = server.getConnectors()[0];
```

Comment:
hm but is that when the 0 gets resolved to the actual local port it bound to?  Or does server.start() guarantee that?